### PR TITLE
LPS-123084 Create utility class SegmentsEntryRetriever (that encapsulates common code for obtaining segments)

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

The main goal of this pull is to create a utility class that contains the common code between SegmentsServicePreAction and SegmentsEntryRoleContributor.

This is the first pull to merge into master the Performance Issue fix with Segmentation (for the full solution take a look to https://github.com/liferay-tango/liferay-portal/pull/505)

**How to test it**

All segments and experience tests should work with this pull.

Also specific Integration tests have been provided for the new utility class.

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci.test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Forwarding, since this is a really small pull, and we need to receive Brian's feedback as soon as possible.

**Comment by cgoncas** [comment]
ci:forward

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/internal/events/test/SegmentsServicePreActionTest.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+		_serviceTracker.close();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testProcessLifecycleEvent() throws Exception {
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/internal/events/test/SegmentsServicePreActionTest.java:99]: Hey @cgoncas, what's the point of this suit of tests? Is it a way of documenting the existing use cases and the ones that are intended to fail in the future? For example, a request without a layout in its context does NOT need to calculate segments

Comment (cgoncas) [reply modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/internal/events/test/SegmentsServicePreActionTest.java:99]: Yeahh!!, that it the reason that every test use a specific layout.

Comment (darquesdev) [reply modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/internal/events/test/SegmentsServicePreActionTest.java:99]: Awesome ❤️

**Pull request closed without merging**